src/primitives.coffee

```diff
@@ -85,6 +85,19 @@ class Collection extends ElementType
 
   add: (val) -> @push val
 
+  findElements: (cond) ->
+    elements = @content.reduce (results, el) ->
+      return results.concat([el.findElements(cond)]) if el.elementType() in ['array', 'object']
+      return results.concat([el]) if cond(el)
+      return results
+    , []
+    _.flatten elements, true
+
+  find: (cond) ->
```

Comment:
@smizell 

> Would you like to see the style of comments used in Drafter? Has that worked well? I haven't wanted to comment much here as this is so early, it could change quite a bit. But I'd like to add something nice.

Any comments, regardless of the format are what I am looking for, even a human readable explanation will work :)